src/nccl_ofi_rdma.c

```diff
@@ -1407,7 +1407,18 @@ static inline int process_completions(struct fi_cq_tagged_entry *cq_entry,
 				return ncclInternalError;
 			}
 
-			if (IS_CONN_RESP_MSG_TYPE(cq_entry[comp_idx].tag) && (comp_flags & FI_RECV)) {
+			if (req->type == NCCL_OFI_RDMA_SEND_CONN) {
+				assert(req->comm->type == NCCL_NET_OFI_SEND_COMM);
```

Comment:
why not move this down under line 1413 and make it `assert(s_comm->type == NCCL_NET_OFI_SEND_COMM);`?

Reply:
To me it makes more sense to assert that we are casting to the correct type before we actually do the cast?

Reply:
I guess I don't see the difference, but if you have strong feelings that's fine.

Reply:
Ack, left as is.